### ISSUE TITLE
Can i able to use WebRTC in Swift Project ?

### ISSUE THREAD

**Issue by indrajitsinh** [ISBX/apprtc-ios#39]
i am new to IOS development and i just recently learned swift programming language and i am not familiar with objective C 

i know there is a feature called bridging header that allows you to use objective c code in Swift project
but as i am new hop someone suggest me what should i do ?

Thank you

**Comment by Xioshock** [comment]
I haven't tried this this project yet, but if you are targeting iOS 8+ and have updated CocoaPods, you don't need bridging header. Just type use_frameworks! at the top of your Podfile. It will create a framework from this library and you can then import it in your swift file, same as UIKit is imported

**Comment by SeanChenU** [comment]
I am using pod 0.39.0 and encountering this issue when `pod install`
The 'Pods-XXX-iOS' target has transitive dependencies that include static binaries: (/Users/User/XXX-ios/Pods/libjingle_peerconnection/libjingle_peerconnection/libWebRTC.a)

**Comment by coolwr** [comment]
Yes, the project should work inside your Swift Project. However, because of the static library dependency you'll need to manually integrate into your project or create a framework per @Xioshock 

The transitive dependency error seems to e an issue with the recent version of cocoa pods. We'll need to investigate this further to find a resolution.

**Comment by SeanChenU** [comment]
Hi @coolwr thanks for replying. Do you mind give us a little hint on how to manually integrate pod into project or create a framework? Any links or keywords are appreciated

**Comment by SeanChenU** [comment]
Here are my steps to integrate APPRTC into my project:

1. Remove: `pod 'APPRTC'` and add them: `pod 'libjingle_peerconnection' ` `pod 'SocketRocket'` in Podfile
2. Copy from Lib/ folder from APPRTC project into my project
3. Add these frameworks and libraries into project: "QuartzCore", "OpenGLES", "CoreGraphics", "CoreVideo", "CoreMedia", "CoreAudio", "AVFoundation", "AudioToolbox", "GLKit", "CFNetwork", "Security", "sqlite3", "stdc++.6", "icucore", "c++"
4. Add related headers in my case `#import <libjingle_peerconnection/RTCEAGLVideoView.h>` `#import "ARDAppClient.h"`  `#import "RTCPeerConnectionFactory.h"` into Bridging-Header.h
5. Build!

**Comment by coolwr** [comment]
Awesome. Thanks for posting the instructions! Would be great if someone could help investigate how to fix the transitive dependency error in cocoa pods so the integration would be easier.

**Comment by cmorell** [comment]
Hi! yes I also did the same like @SeanChenU but I really like how to add appRTC pod or how to create ones in order to has AppRTC client as a Pod and be usefull using swift project. Can anyone did it before?

**Comment by saiteja7** [comment]
what is the meanning of 2nd step @SeanChenU

**Comment by cmorell** [comment]
@saiteja7 the meaning of 2nd step in @SeanChenU comment is that we need to copy in our project the classes that contains AppRTC/Lib folder:

ARDAppClient.(h/m)
ARDMessageResponse.(h/m)
ARDRegisterResponse.(h/m)
ARDSignalingMessage.(h/m)
ARDUtilities.(h/m)
ARDWebSocketChannel.(h/m)
RTCICECandidate+JSON.(h/m)
RTCICEServer+JSON.(h/m)
RTCMediaConstraints.(h/m)
RTCSessionDescription.(h/m)

**Comment by saiteja7** [comment]
okay thanq

**Comment by sonysum** [comment]
Hi,
@SeanChenU  i follow your solution but i can not call  renderFrame, 
Ex: [self.localView renderFrame:nil]; // error class has no men=mber 'renderFrame'
any solution?

**Comment by techwinlabs** [comment]
i am having the same issue in a swift project @cmorell , 
https://gist.github.com/techwinlabs/07c416362ae7b4dd3ea78eacd4a428cf.js
Any solution found for this?

**Comment by Carlos77** [comment]
I have the same problem can i have more informations for how to solve this issues 


WorkSpace/Test2/platforms/ios/Test2/Plugins/com.onsip.cordova/libwebrtc_base.a
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_RTCEAGLVideoView", referenced from:
      objc-class-ref in PhoneRTCPlugin.o
  "_OBJC_CLASS_$_RTCICEServer", referenced from:
      objc-class-ref in PhoneRTCPlugin.o
  "_OBJC_CLASS_$_RTCICECandidate", referenced from:
      objc-class-ref in PhoneRTCDelegate.o
  "_OBJC_CLASS_$_RTCMediaConstraints", referenced from:
      objc-class-ref in PhoneRTCDelegate.o
  "_OBJC_CLASS_$_RTCPair", referenced from:
      objc-class-ref in PhoneRTCDelegate.o
  "_OBJC_CLASS_$_RTCVideoCapturer", referenced from:
      objc-class-ref in PhoneRTCDelegate.o
  "_OBJC_CLASS_$_RTCSessionDescription", referenced from:
      objc-class-ref in PhoneRTCDelegate.o
  "_OBJC_CLASS_$_RTCPeerConnectionFactory", referenced from:
      objc-class-ref in PhoneRTCDelegate.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by hindalsayyar** [comment]
@Carlos77 did you found solution?
I am getting the same error.
